docs/sphinx/extensions/sphinx-hpx.py

```
@@ -1,11 +1,44 @@
+                _hpx_api_anchors = json.load(f)
+        except (OSError, ValueError):
+            return {}
+    return _hpx_api_anchors
+
+
+class HPXAPIRole(XRefRole):
+    def run(self):
+        anchors = _load_hpx_api_anchors(self.env.app)
+        anchor = anchors.get(self.target)
+        if anchor is None:
+            logger.warning('hpx-api: no generated page for %r' % target)
```

Ruff (0.16.0)

[error] Undefined name `target`

(F821)